lib/OrdinaryDiffEqTaylorSeries/src/alg_utils.jl

```diff
@@ -0,0 +1,55 @@
+alg_order(::ExplicitTaylor2) = 2
+alg_stability_size(alg::ExplicitTaylor2) = 1
+
+alg_order(::ExplicitTaylor{P}) where {P} = P
+alg_stability_size(alg::ExplicitTaylor) = 1
+
+JET_CACHE = IdDict()
```

Comment:
`const`?

Reply:
also thread safety?

Reply:
How to make this thread-safe? Do I need to switch to other data structures

Reply:
if you're going to use a global cache, you need to synchronize mutation to it. Alternatively, is there a way to make the cache local?

Reply:
rewriting you are guaranteed to have the same function right? So I don't think it's unsafe. What might happen is you might compile more times than you need to, but that would still be correct.

Reply:
no. If you have concurent modification of an IDDict without synchronization, that's a data race (which is UB). You can get torn writes, or other arbitrarily wrong results getting written.

Reply:
Okay torn writes would be bad. So it just needs a lock on write?

Reply:
I think the best approach would be to make the cache non-global. As long as you have a separate cache per function (which should be totally fine) there won't be multithreading to cause problems.